File: leetcode/0013. Roman to Integer/Solution.java
class Solution {
    public int romanToInt(String s) {
        // Populate the lookup table.
        Map<Character, Integer> map = new HashMap<Character, Integer>();
        map.put('I', 1);
        map.put('V', 5);
        map.put('X', 10);
        map.put('L', 50);
        map.put('C', 100);
        map.put('D', 500);
        map.put('M', 1000);

        int length = s.length();
        int result = 0;
        int first = 0;
        int second = map.get(s.charAt(0));
        for(int i = 0; i < length - 1; i++) {
            first = map.get(s.charAt(i));
            second = map.get(s.charAt(i + 1));
            if(first < second) {
                result += second - first;
                i++;
                second = 0;
                if (i + 2 == length) {
                    second = map.get(s.charAt(i + 1));
                }
            }
            else {
                result += first;
            }
        }
        result += second;
        return result;
    }
}

File: leetcode/0013. Roman to Integer/Solution.py
class Solution(object):
    def romanToInt(self, s):
        """
        :type s: str
        :rtype: int
        """
        mappings = {"I": 1, "V": 5, "X": 10, "L": 50, "C": 100, "D": 500, "M": 1000}
        result = 0

        i = len(s) - 1
        while i >= 0:
            value = mappings[s[i]]
            if i != 0 and mappings[s[i - 1]] < value:
                value -= mappings[s[i - 1]]
                i -= 1
            result += value
            i -= 1
        return result
        
